rust/sample_applications/sample_connector_scaffolding/src/main.rs

```diff
@@ -0,0 +1,517 @@
+    // asset_ready_watcher_tx.send_if_modified(send_if_modified_fn(false));
+
+    // Receive asset updates and dataset creation notifications
+    loop {
+        match asset_client.recv_notification().await {
+            ClientNotification::Updated => {
+                log::info!("{asset_log_identifier} Asset update notification received");
+
+                // IMPLEMENT: Add custom asset update/validation logic here
+
+                // For this example, we will assume that the asset specification is OK. Modify this to report any errors
+                match asset_client.report_status(Ok(())).await {
+                    Ok(()) => {
+                        log::debug!("{asset_log_identifier} Asset status reported as OK");
+                    }
+                    Err(e) => {
+                        log::error!("{asset_log_identifier} Failed to report asset status: {e}");
+                    }
+                }
+                // Notify any datasets that the asset is in a ready state (this notification will only be sent if that wasn't already true)
+                asset_ready_watcher_tx.send_if_modified(send_if_modified_fn(true));
+                // If there was an error, notify any lower components that the asset is not in a ready state while we wait for another update
+                // asset_ready_watcher_tx.send_if_modified(send_if_modified_fn(false));
+            }
+            ClientNotification::Created(dataset_client) => {
+                let dataset_log_identifier = {
+                    let dataset_ref = dataset_client.dataset_ref();
+                    format!("{asset_log_identifier}[DS: {}]", dataset_ref.dataset_name)
+                };
+                log::info!("{dataset_log_identifier} Dataset created");
+
+                // Handle the new dataset
+                tokio::task::spawn(handle_dataset(
+                    dataset_log_identifier,
+                    dataset_client,
+                    asset_ready_watcher_tx.subscribe(),
+                    device_endpoint_ready_watcher_rx.clone(),
+                ));
+            }
+            ClientNotification::Deleted => {
+                log::info!(
+                    "{asset_log_identifier} Asset deleted notification received, ending asset handler"
+                );
+                // The asset ready state does not need to be updated here because all datasets will also get deleted
+                break;
+            }
+        }
+    }
+}
+
+/// Handles sampling of data from the dataset.
+///
+/// # Arguments
+/// * `dataset_log_identifier` - A string identifier for the dataset, used for logging.
+/// * `dataset_client` - The dataset client.
+/// * `asset_ready_watcher_rx` - A watcher for the asset readiness state.
+/// * `device_endpoint_ready_watcher_rx` - A watcher for the device endpoint readiness state.
+#[allow(unused_assignments)] // IMPLEMENT: Remove once variables are being used
```

Comment:
Why not just use underscore prefixes?

Reply:
because it's just one of the assignments that would be inside an if statement in a real implementation. There is a change I can make to remove this though, I'll do that instead